operator/pkg/gateway-api/tlsroute.go

```diff
@@ -42,25 +43,56 @@ func newTLSRouteReconciler(mgr ctrl.Manager) *tlsRouteReconciler {
 func (r *tlsRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &gatewayv1alpha2.TLSRoute{}, backendServiceIndex,
 		func(rawObj client.Object) []string {
-			hr, ok := rawObj.(*gatewayv1alpha2.TLSRoute)
+			route, ok := rawObj.(*gatewayv1alpha2.TLSRoute)
 			if !ok {
 				return nil
 			}
 			var backendServices []string
+			for _, rule := range route.Spec.Rules {
+				for _, backend := range rule.BackendRefs {
+					namespace := helpers.NamespaceDerefOr(backend.Namespace, route.Namespace)
+					backendServiceName, err := helpers.GetBackendServiceName(r.Client, namespace, backend.BackendObjectReference)
+					if err != nil {
+						log.WithFields(logrus.Fields{
+							logfields.Controller: "tlsRoute",
+							logfields.Resource:   client.ObjectKeyFromObject(rawObj),
+						}).WithError(err).Error("Failed to get backend service name")
```

Comment:
in case of an error, we need to `continue` without adding the service to the slice